# Fixes two tab-nick-completion bugs:
# 1. Crash when you type space-tab
# 2. Critical glib warning when completion in a dialog-tab.
--- src/fe-gtk/fkeys.c.orig	Thu Aug  5 07:49:06 2004
+++ src/fe-gtk/fkeys.c	Mon Aug 16 22:54:58 2004
@@ -1427,7 +1427,7 @@
 			skip_len++;
 		}
 		else
-			cursor_pos = g_utf8_pointer_to_offset(text, g_utf8_offset_to_pointer(ch, 1));
+			return 2;
 	}
 
 	comp = skip_len;
@@ -1608,7 +1608,8 @@
 						list = list->next;
 					}
 					PrintText (sess, buf);
-					g_completion_free(gcomp);
+					if (gcomp)
+						g_completion_free(gcomp);
 					return 2;
 				}
 				/* Only one matching entry */
